### PR TITLE
62 make game machin independent

### PR DESCRIPTION
**Pull request by Ozymadias**

### CODE REVIEW

File: battleship-client/src/main/java/com/epam/solid/nie/client/Main.java [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 public class Main {
     public static void main(String[] args) {
-        ShipClient shipServer = new ShipSocketClient();
+        System.out.println("server ip is: " + (args.length > 0 ? args[0] : ""));
+        ShipClient shipServer = new ShipSocketClient(args[0]);
```
Comment (rivalez) [inline battleship-client/src/main/java/com/epam/solid/nie/client/Main.java]: nazwa zmiennej shipClient zamiast shipServer

File: battleship-server/src/main/java/com.epam.solid.nie/server/CommunicationProtocol.java [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 public class CommunicationProtocol implements Protocol {
     @Override
-    public String processInput(String inputString) {
+    public String processInput(String inputString, int id) {
```
Comment (rivalez) [inline battleship-server/src/main/java/com.epam.solid.nie/server/CommunicationProtocol.java]: what id stands for? change name to something more meaningful / or write java doc for this method

File: battleship-server/src/main/java/com.epam.solid.nie/server/ServerThread.java [hunk 3]
```diff
@@ -8,9 +8,11 @@
 
 class ServerThread extends Thread{
     private final Socket clientSocket;
+    private final int id;
```
Comment (rivalez) [inline battleship-server/src/main/java/com.epam.solid.nie/server/ServerThread.java]: the same thing like in Protocol interface. Why we need this id?

File: battleship-server/src/main/java/com.epam.solid.nie/server/ShipSocketServer.java [hunk 4]
```diff
@@ -6,16 +6,25 @@
...
+    private final String ip;
+
+    ShipSocketServer(String ip) {
+        this.ip = ip;
+    }
 
     @Override
     public void run() {
-        try (ServerSocket serverSocket = new ServerSocket(portNumber, 0, InetAddress.getLoopbackAddress())) {
+        int portNumber = 8080;
```
Comment (rivalez) [inline battleship-server/src/main/java/com.epam.solid.nie/server/ShipSocketServer.java]: private final int variable was good idea

File: battleship-server/src/main/java/com.epam.solid.nie/server/CommunicationProtocol.java (hunk 2 above)
Comment (Ozymadias) [reply battleship-server/src/main/java/com.epam.solid.nie/server/CommunicationProtocol.java]: removed in last commit

File: battleship-server/src/main/java/com.epam.solid.nie/server/ServerThread.java (hunk 3 above)
Comment (Ozymadias) [reply battleship-server/src/main/java/com.epam.solid.nie/server/ServerThread.java]: removed in last commit

File: battleship-server/src/main/java/com.epam.solid.nie/server/ShipSocketServer.java (hunk 4 above)
Comment (Ozymadias) [reply battleship-server/src/main/java/com.epam.solid.nie/server/ShipSocketServer.java]: so should it be a field in your opinion?

File: battleship-client/src/main/java/com/epam/solid/nie/client/Main.java (hunk 1 above)
Comment (Ozymadias) [reply battleship-client/src/main/java/com/epam/solid/nie/client/Main.java]: changed in last commit

**Pull request merged**